bluepysnap/circuit_ids.py

```diff
@@ -58,7 +58,11 @@ def __init__(self, index, sort_index=True):
     @property
     def index_schema(self):
         """Return an empty index with the same names of the wrapped index."""
-        return pd.MultiIndex.from_tuples([], names=self.index.names)
+        # Since pandas v2.1.0, need to set dtypes explicitly
+        levels = [np.array([], dtype=dtype) for dtype in self.index.dtypes]
+        codes = len(levels) * [[]]
+
+        return pd.MultiIndex(levels=levels, codes=codes, names=self.index.names)
```

Comment:
I see that the result of `pd.MultiIndex.from_tuples([], names=["a", "b"])` is the same in pandas 2.0.3 and 2.1.0 (i.e. `idx.dtypes` are always objects).
I suspect that something changed in `assert_index_equal()` or `assert_frame_equal()`.

Regardless, instead of manually building a new index, you could just return:
```python
return self.index[:0]
```

Reply:
> I see that the result of pd.MultiIndex.from_tuples([], names=["a", "b"]) is the same in pandas 2.0.3 and 2.1.0 (i.e. idx.dtypes are always objects).

Damnit. I'll take another look to get to the bottom of this.

> Regardless, instead of manually building a new index, you could just return

Holy cow, I tried just about everything but not that. 🤦  Thanks!

Reply:
AAMOF, the position of the multi-index having dtypes of object does not matter: as long as there's any of the indices having dtype=object, the concatenated index will have that dtype.

Reply:
Thanks for the investigation. Another thing that I noticed in your example:

```python
>>> dfs[0]['c'].dtype
dtype('float64')
>>> dfs[1]['c'].dtype
dtype('int64')
>>> df['c'].dtype
dtype('float64')
```

To avoid casting to float64, in cases like this one, we may need to ensure that the dtype of `dfs[0]['c']` is set to `int64` (or skip empty dataframes when concatenating, but it seems less clean)

Reply:
I believe that the setting of dtypes is covered in https://github.com/BlueBrain/snap/blob/master/bluepysnap/network.py#L169C10-L169C10

Either way, the dataframes will no longer be concatenated after #217 , nor will the `index_schema` be used.